include/itkPyTorchImage.h

```diff
@@ -134,106 +134,127 @@ class PyTorchImage : public Image< TPixel, VImageDimension >
     using PixelType = CovariantVector< TScalar, NVectorComponent >;
     using ValueType = typename PixelType::ValueType;
     static constexpr SizeValueType NumberOfComponents = NVectorComponent;
+    using PixelTypeIsVectorType = int;
     static PixelType pixelInstance( unsigned numberOfComponents ) { return PixelType {}; }
     };
   using ValueType = typename PixelHelper< PixelType >::ValueType;
+
+  static constexpr unsigned int ImageDimension = VImageDimension;
   static constexpr at::ScalarType PyTorchValueType = c10::impl::CPPTypeToScalarType< ValueType >::value;
-  static constexpr SizeValueType NumberOfComponents = PixelHelper< PixelType >::NumberOfComponents;
-  static auto pixelInstance = PixelHelper< PixelType >::pixelInstance;
+  static constexpr auto pixelInstance = PixelHelper< PixelType >::pixelInstance;
+
+  // A pixel value may be a basic type (integer, real), or a class (RGB, RGBA, Vector, CovariantVector) whose value
+  // types are, recursively pixel types.  DimensionHelper can handle all cases that PixelHelper knows about.
+  // For the case that the TPixelType is not known by PixelHelper:
+  template< typename TPixelType, typename = void >
+  struct DimensionHelper
+    {
+    using PixelType = TPixelType;
+    };
+  // For the case that the TPixelType is known by PixelHelper to be a scalar type:
+  template< typename TPixelType >
+  struct DimensionHelper< TPixelType, myVoid_t< typename PixelHelper< TPixelType >::PixelTypeIsScalarType > >
+    {
+    using PixelType = TPixelType;
+    using DeepScalarType = PixelType;
+    static void AppendSizes( std::vector< SizeValueType > &size ) {/* Nothing to append */}
+    };
+  // For the case that the TPixelType is known by PixelHelper to be a vector type:
+  template< typename TPixelType >
+  struct DimensionHelper< TPixelType, myVoid_t< typename PixelHelper< TPixelType >::PixelTypeIsVectorType > >
+    {
+    using PixelType = TPixelType;
+    using DeepScalarType = typename DimensionHelper< typename PixelHelper< PixelType >::ValueType >::DeepScalarType;
+    static void AppendSizes( std::vector< SizeValueType > &size )
+      {
+      size.push_back( PixelHelper< PixelType >::NumberOfComponents );
+      // Recurse
+      DimensionHelper< typename PixelHelper< PixelType >::ValueType >::AppendSizes( size );
+      }
+    };
+  using DeepScalarType = typename DimensionHelper< PixelType >::DeepScalarType;
 
   /** Allocate CPU and GPU memory space */
   virtual void Allocate( bool initialize = false ) override;
 
-  virtual void AllocateGPU() override;
+  virtual void AllocateGPU();
 
   virtual void Initialize() override;
 
-  virtual void FillBuffer( const TPixel &value ) override;
+  virtual void FillBuffer( const TPixel &value ) /* `override` fails because not virtual in itkImage!!! */;
```

Comment:
Then it probably should not be `virtual` here either?

Reply:
I agree that they should be consistent.  I would appreciate knowing why they are not virtual in Image<...>.